Src/IronPython/Runtime/Operations/FloatOps.cs

```diff
@@ -1017,6 +1017,46 @@ public static partial class DoubleOps {
                     throw PythonOps.ValueError("__setformat__() argument 1 must be 'double' or 'float'");
             }
         }
+
+        public static double __round__(double self) {
+            // as of Python 3 rounding is to the nearest even number, not away from zero
+            return __round__(self, 0);
+        }
+
+        public static double __round__(double self, int ndigits) {
+            if (double.IsNaN(self)) {
```

Comment:
That overload calls into this one so we don't need to replicate the checks. There should be tests for these cases.

Reply:
Having nan/inf in this overload is valid. `float('nan').__round__(0)`

Reply:
> Having nan/inf in this overload is valid. `float('nan').__round__(0)`

I don't quite follow you. Are you saying it's OK as it is but pointing out that I missed out the tests for the `ndigits` case? I've added them now.

Reply:
The `double.IsNaN` and `double.IsInfinity` checks are only valid in the case where `ndigits` is undefined since that overload is expected to return an `int` (as you noted in https://github.com/IronLanguages/ironpython3/issues/659). As soon as you specify the `ndigits` argument it returns a `double` so `nan` and `inf` become valid values.
```Python
try:
    float('nan').__round__()
    assert False
except ValueError:
    pass

assert math.isnan(float('nan').__round__(0)) # this one shouldn't throw
```

Reply:
OK, got you.